### PR TITLE
Fix ist0103 msg incorrect displaying when injection label is set to false

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
```
Warning [IST0103] (Pod default/reviews-v2-5b5dc658c5-z4c9s) The pod default/reviews-v2-5b5dc658c5-z4c9s is missing the Istio proxy. This can often be resolved by restarting or redeploying the workload.
Error: Analyzers found issues when analyzing namespace: default.
```
```
kubectl get deploy reviews-v2 -oyaml | grep sidecar. -B2
      labels:
        app: reviews
        sidecar.istio.io/inject: "false"
```

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [x] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pkg/config/analysis/analyzers/testdata/injection.yaml [hunk 1]
```diff
@@ -99,3 +99,17 @@ spec:
...
     name: server
+---
+# Pod that explicitly disables injection with label. Should not generate a warning.
+apiVersion: v1
+kind: Pod
+metadata:
+  name: podinjectiondisabled-label
+  namespace: default
+  labels:
+    sidecar.istio.io/inject: "false"
```
Comment (GregHanson) [inline pkg/config/analysis/analyzers/testdata/injection.yaml:110]: do we support this as a label?  I thought `sidecar.istio.io/inject` was supported as an annotation ([doc](https://istio.io/latest/docs/reference/config/annotations/)).  In which case there is a test already covering the annotation on line 76

**Review by GregHanson — CHANGES_REQUESTED** [review]

File: pkg/config/analysis/analyzers/testdata/injection.yaml (hunk 1 above)
Comment (hanxiaop) [reply pkg/config/analysis/analyzers/testdata/injection.yaml:110]: @GregHanson See https://istio.io/latest/docs/setup/additional-setup/sidecar-injection/#controlling-the-injection-policy. If both are set, we prefer the label.

Comment (GregHanson) [reply pkg/config/analysis/analyzers/testdata/injection.yaml:110]: @hanxiaop thanks for clarifying, I see the annotation and label logic in `/pkg/kube/inject/inject.go`.

**Review by GregHanson — APPROVED** [review]

**Pull request merged**